File: src/main/java/hu/inf/unideb/model/Car.java
package hu.inf.unideb.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "cars")
public class Car {

    @Id
    @EqualsAndHashCode.Include
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(unique = true)
    private String vin;

    private String make;

    private String model;

    private int year;

    @Enumerated(EnumType.STRING)
    private Fuel fuel;

    @Enumerated(EnumType.STRING)
    private Transmission transmission;

    private int engine_size;

    private List<String> images_src;

    private int price;

    private String description;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonBackReference
    private User user;

}

File: car_marketplace_project_fe/src/Types/Car.ts

import type { User } from "./User.ts";

export type Fuel = "PETROL" | "DIESEL" | "HYBRID" | "ELECTRIC";
export type Transmission = "MANUAL" | "AUTOMATIC";

export interface Car {
    id: number;
    vin: string;
    make: string;
    model: string;
    year: number;
    fuel: Fuel;
    transmission: Transmission;
    engine_size: number;
    images_src?: string[];
    price: number;
    description?: string;
    user?: User;
    username?: string;
}

export type carType = {
    id: number;
    vin: string;
    make: string;
    model: string;
    year: number;
    fuel: Fuel;
    transmission: Transmission;
    engine_size: number;
    images_src?: string[];
    price: number;
    description?: string;
    user?: User;
    username?: string;
};


export type carRequest = {
    vin: string;
    make: string;
    model: string;
    year: number;
    fuel: Fuel;
    transmission: Transmission;
    engine_size: number;
    images_src?: string[];
    price: number;
    description?: string;
};
